engine/access/rpc/backend/backend_transactions_test.go

```diff
@@ -0,0 +1,329 @@
+		suite.state.On("AtBlockID", block.ID()).Return(snap, nil).Once()
+
+		suite.transactions.
+			On("ByID", tx.ID()).
+			Return(nil, storage.ErrNotFound)
+
+		suite.blocks.
+			On("ByID", block.ID()).
+			Return(&block, nil).
+			Once()
+
+		receipt := unittest.ExecutionReceiptFixture()
+		identity := unittest.IdentityFixture()
+		identity.Role = flow.RoleExecution
+
+		l := flow.ExecutionReceiptList{receipt}
+
+		suite.receipts.
+			On("ByBlockID", block.ID()).
+			Return(l, nil)
+
+		backend := New(
+			suite.state,
+			suite.colClient,
+			nil,
+			suite.blocks,
+			nil,
+			nil,
+			suite.transactions,
+			suite.receipts,
+			nil,
+			suite.chainID,
+			metrics.NewNoopCollector(),
+			nil,
+			false,
+			DefaultMaxHeightRange,
+			nil,
+			nil,
+			suite.log,
+			DefaultSnapshotHistoryLimit,
+			nil,
+			*suite.communicator,
+			false,
+		)
+		res, err := backend.GetTransactionResult(context.Background(), tx.ID(), block.ID(), coll.ID())
+		suite.Require().NoError(err)
+		suite.Require().Equal(res.Status, flow.TransactionStatusUnknown)
+	})
+}
+
+func (suite *Suite) TestGetTransactionResultReturnsTransactionError() {
+	suite.WithPreConfiguredState(func(snap protocol.Snapshot) {
+		block := unittest.BlockFixture()
+		tbody := unittest.TransactionBodyFixture()
+		tx := unittest.TransactionFixture()
+		tx.TransactionBody = tbody
+
+		coll := flow.CollectionFromTransactions([]*flow.Transaction{&tx})
+
+		suite.transactions.
+			On("ByID", tx.ID()).
+			Return(nil, fmt.Errorf("some other error"))
+
+		suite.blocks.
+			On("ByID", block.ID()).
+			Return(&block, nil).
+			Once()
+
+		receipt := unittest.ExecutionReceiptFixture()
+		identity := unittest.IdentityFixture()
+		identity.Role = flow.RoleExecution
+
+		suite.state.On("AtBlockID", block.ID()).Return(snap, nil).Once()
+
+		l := flow.ExecutionReceiptList{receipt}
+
+		suite.receipts.
+			On("ByBlockID", block.ID()).
+			Return(l, nil)
+
+		backend := New(
+			suite.state,
+			suite.colClient,
+			nil,
+			suite.blocks,
+			nil,
+			nil,
+			suite.transactions,
+			suite.receipts,
+			nil,
+			suite.chainID,
+			metrics.NewNoopCollector(),
+			nil,
+			false,
+			DefaultMaxHeightRange,
+			nil,
+			nil,
+			suite.log,
+			DefaultSnapshotHistoryLimit,
+			nil,
+			*suite.communicator,
+			false,
+		)
+		_, err := backend.GetTransactionResult(context.Background(), tx.ID(), block.ID(), coll.ID())
+		suite.Require().Equal(err, status.Errorf(codes.Internal, "failed to find: %v", fmt.Errorf("some other error")))
+
+	})
+}
+
+func (suite *Suite) TestGetTransactionResultReturnsValidTransactionResult() {
+	suite.WithPreConfiguredState(func(snap protocol.Snapshot) {
+		block := unittest.BlockFixture()
+		tbody := unittest.TransactionBodyFixture()
+		tx := unittest.TransactionFixture()
+		tx.TransactionBody = tbody
+
+		coll := flow.CollectionFromTransactions([]*flow.Transaction{&tx})
+
+		suite.transactions.
+			On("ByID", tx.ID()).
+			Return(nil, storage.ErrNotFound)
+
+		suite.blocks.
```

Comment:
should this be called in this case? best to remove mock calls that aren't used so the tested path is clearer.

Reply:
removed

Reply:
entire test case has been refactored, kindly see here https://github.com/onflow/flow-go/pull/4598/files#diff-3106a362ce9111d9a63804ec67a9afbb871eb56bf354cf65bcb9aaf37d98f932